design/pipeline-resources.md

```diff
@@ -0,0 +1,199 @@
+# Resources in YAML
+
+Any external service that is consumed as part of your pipeline is a resource. 
+
+An example of a resource can be another CI/CD pipeline that produces artifacts (say Azure pipelines, Jenkins etc.), code repositories (GitHub, Azure Repos, Git), container image registries (ACR, Docker hub etc.) or package feeds (Azure artifact feed, Artifactor etc.).  
+
+## Why resources?
+
+Resources are defined at one place and can be consumed anywhere in your pipeline. Resources provide you the full traceablity of the services consumed in your pipeline including the branch, version, tags, associated commits and work-items. You can fully automate your DevOps workflow by subscribing to trigger events on your resources.
+
+Resources in YAML represent sources of types pipelines, repositories, containers and packages.
+
+
+### Schema
+
+```yaml
+resources:
+  pipelines: [ pipeline ]  
+  repositories: [ repository ]
+  containers: [ container ]
+  packages: [ package ]
+```
+
+---
+
+## Resources: `pipelines`
+
+If you have a pipeline that produces artifacts, you can consume the artifacts by defining a `pipelines` resource. A pipeline can be another Azure DevOps pipeline or any external pipelines like Jenkins etc.
+
+
+### Schema
+
+```yaml
+resources:        # types: pipelines | repositories | containers | packages
+  pipelines:
+  - pipeline: string  # identifier for the pipeline resource
+    type: enum  # type of the pipeline source like azurePipelines, Jenkins etc. 
+    connection: string  # service connection to connect to the source
+    source:  
+      name: string  # source defintion of the pipeline including the project i.e. projectName/definition
+      version: string  # version to pick the artifact, optional; defaults to Latest
+      branch: string  # branch to pick the artiafct, optional; defaults to master branch
+      tags: string # picks the artifacts on from the pipeline with given tag, optional; defaults to no tags.
+```
+
+The inputs inside `source` can change based on the pipeline type defined. The above schema is for the `type`: azurePipelines.
+
+
+### Examples
+
+If you need to consume another azurePipelines from the current project and you dont require setting branch, version and tags info, this can be shortened to:
+
+```yaml
+resources:         
+  pipelines:
+  - pipeline: SmartHotel      
+    type: azurePipelines
+    source: SmartHotel-CI  # name of the pipeline source definition
```

Comment:
Is this valid YAML? I don't think so but @vtbassmatt is a better judge than I am.

Reply:
yes, it is a valid YAML. Please refer to the image below.

![yaml syntax](https://user-images.githubusercontent.com/25578249/49072462-48f62400-f256-11e8-8f4e-0e60f06a1cb2.PNG)